### PR TITLE
Implement transaction scrubbing for First Data e4

### PR DESCRIPTION
**Pull request by girasquid**
@j-mutter @ivanfer for review

### CODE REVIEW

File: test/unit/gateways/firstdata_e4_test.rb [hunk 1]
```diff
@@ -163,8 +163,76 @@ def test_add_swipe_data_with_creditcard
...
   private
 
+  def pre_scrubbed
+    <<-PRE_SCRUBBED
+      opening connection to api.demo.globalgatewaye4.firstdata.com:443...
+      opened
+      starting SSL for api.demo.globalgatewaye4.firstdata.com:443...
+      SSL established
+      <- "POST /transaction/v11 HTTP/1.1\r\nContent-Type: application/xml\r\nAccepts: application/xml\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: Ruby\r\nConnection: close\r\nHost: api.demo.globalgatewaye4.firstdata.com\r\nContent-Length: 593\r\n\r\n"
+      <- "<?xml version=\"1.0\" encoding=\"UTF-8\"?><Transaction><ExactID>REDACTED</ExactID><Password>REDACTED</Password><Transaction_Type>00</Transaction_Type><DollarAmount>1.00</DollarAmount><Card_Number>4242424242424242</Card_Number><Expiry_Date>0916</Expiry_Date><CardHoldersName>Longbob Longsen</CardHoldersName><CardType>Visa</CardType><VerificationStr1>1234 My Street|K1C2N6|Ottawa|ON|CA</VerificationStr1><CVD_Presence_Ind>1</CVD_Presence_Ind><VerificationStr2>123</VerificationStr2><Reference_No>1</Reference_No><Reference_3>Store Purchase</Reference_3><CAVV/><XID/><Ecommerce_Flag/></Transaction>"
```
Comment (ivanfer) [inline test/unit/gateways/firstdata_e4_test.rb]: what does the password ``` REDACTED ``` here? xml lib?

Comment (girasquid) [inline test/unit/gateways/firstdata_e4_test.rb]: There are no credentials in the fixtures file, so I needed to run this with credentials that I can't leave in the code publicly.

File: test/remote/gateways/remote_firstdata_e4_test.rb [hunk 2]
```diff
@@ -136,10 +136,14 @@ def test_refund_with_track_data
...
-  
+
     assert response = @gateway.refund(50, purchase.authorization)
     assert_success response
     assert_match(/Transaction Normal/, response.message)
     assert response.authorization
   end
+
+  def test_dump_transcript
+    # dump_transcript_and_fail(@gateway, @amount, @credit_card, @options)
```
Comment (ivanfer) [inline test/remote/gateways/remote_firstdata_e4_test.rb]: A skip message would be nice here to remove the ambiguity of an empty test.

Comment (girasquid) [inline test/remote/gateways/remote_firstdata_e4_test.rb]: Changed this to be `# See firstdata_e4_test.rb for an example of a scrubbed transcript`.

Comment (ivanfer) [inline test/remote/gateways/remote_firstdata_e4_test.rb]: was thinking of ``` skip("scrubbing already done " ) ```, but not a big deal.

File: test/unit/gateways/firstdata_e4_test.rb (hunk 1 above)
Comment (ivanfer) [inline test/unit/gateways/firstdata_e4_test.rb]: Seems sensitive, do we want to scrub those?

**Comment by ivanfer** [comment]
questions, otherwise :cactus:

File: test/unit/gateways/firstdata_e4_test.rb (hunk 1 above)
Comment (girasquid) [inline test/unit/gateways/firstdata_e4_test.rb]: Not PCI sensitive; if we want to scrub those it should be done somewhere else.

Comment (ivanfer) [inline test/unit/gateways/firstdata_e4_test.rb]: :cool:

**Comment by aprofeit** [comment]
:+1:

**Pull request closed without merging**